docs/build_parameters.rst

```diff
@@ -0,0 +1,833 @@
+    - url: https://container-registry.example.com/v2
+      auth: /var/run/secrets/atomic-reactor/v2-registry-dockercfg
+
+    source_registry:
+        url: https://registry.private.example.com
+
+    sources_command: "fedpkg sources"
+
+    required_secrets:
+    - kojisecret
+    - pulpsecret
+    - odcssecret
+    - v2-registry-dockercfg
+
+    worker_token_secrets:
+    - x86-64-worker-1
+    - x86-64-worker-2
+
+    build_json_store: /usr/share/osbs/
+
+
+Secrets
+"""""""
+
+Because the plugin configuration will be rendered at build time (after ``Build``
+object is created), we no longer can select which secrets to mount in container
+build based on which plugins have been enabled. Instead, all the secrets that
+may be needed will be mounted. The **reactor_config** ``ConfigMap`` will define
+the full set of secrets it needs via its **required_secrets** list.
+
+When orchestrator build starts worker builds, it'll use the same set of secrets.
+This requires worker clusters to have the same set of secrets available. For
+example, if **reactor_config** defines::
+
+    required_secrets:
+    - kojisecret
+    - pulpsecret
+
+Secrets named kojisecret and pulpsecret must be available in orchestrator and
+worker clusters. They don't need to have the same value, just the same name.
+
+Secrets needed for communication from orchestrator build to worker clusters are
+defined separately in **worker_token_secrets**. These will not be passed along
+to worker builds.
+
+Atomic Reactor Plugins
+""""""""""""""""""""""
+
+After **reactor_config** is capable of providing environment parameters, various
+atomic reactor plugins will change to retrieve environment parameters from
+**reactor_config** instead of taking those values as their own plugin
+parameters.
+
+To allow a smoother transition, we'll introduce the new arrangement version 6.
+**reactor_config** plugin will provide a helper method so plugins can query the
+current arrangement version in use and decide the source of environment
+parameters. Plugin parameters that are really environment parameters will be
+modified to be optional. Over time, previous arrangement versions will be
+removed, and eventually, these plugin parameters can be removed completely, as
+well as the arrangement version conditional.
+
+
+Arrangement Version 6
+"""""""""""""""""""""
+
+The purpose of this new arrangement version is to easily identify whether or not
+environment parameters are provided by **reactor_config**. The order of plugins
+is not expected to change. However, hard coded, or placeholder, environment
+parameters in **orchestrator_inner** and **worker_inner** json files will
+change.
+
+A new osbs-client configuration **reactor_config_map** will be added to define
+the name of the ``ConfiMap`` object holding **reactor_config**. This
+configuration option will be mandatory for arrangement versions greater than or
+equal to 6. The existing osbs-client configuration **reactor_config_secret**
+will be deprecated.
```

Comment:
Warn if present when arrangement_version >= 6?

Reply:
I'm wondering if we can actually use `reactor_config_map` for any arrangement, and *always* warn that `reactor_config_secret` is deprecated. What do you think?

Reply:
Yes, I like that. Simpler.

Reply:
Changed text to:

> ... will be deprecated (for all arrangements).

Reply:
Thanks. So, the old option name will still be accepted, but osbs-client will issue a diagnostic when it is used.